java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -1514,6 +1514,26 @@ void testStringLocateOffsets() {
     }
   }
 
+  @Test
+  void testSubString() {
+    try (ColumnVector v = ColumnVector.fromStrings("Héllo", "thésé", null,"", "ARé", "strings");
+         ColumnVector e_allParameters = ColumnVector.fromStrings("llo", "ésé", null, "", "é", "rin");
+         ColumnVector e_withoutStop = ColumnVector.fromStrings("llo", "ésé", null, "", "é", "rings");
+         ColumnVector result = v.subString(2, 5);
+         ColumnVector result1 = v.subString(2)) {
+      assertColumnsAreEqual(e_allParameters, result);
+      assertColumnsAreEqual(e_withoutStop,result1);
+    }
+  }
+
+  @Test
+  void testSubStringThrowsException() {
+    assertThrows(AssertionError.class, () -> {
+      try (ColumnVector cv = ColumnVector.fromStrings("Héllo", "thésé", null, "ARé", "tést strings");
+           ColumnVector locate = cv.subString(-2, 6)) {}
```

Comment:
lgtm mostly - Curiosu what happens when end index is negative but not -1

Reply:
Thanks for reviewing @kuhushukla . The behavior would be same for any negative number if assigned to end index.